nixos/tests/vikunja.nix

```diff
@@ -0,0 +1,66 @@
+import ./make-test-python.nix ({ pkgs, lib, ... }: {
+  name = "vikunja";
+
+  meta = with lib.maintainers; {
+    maintainers = [ em0lar ];
+  };
+
+  nodes = {
+    vikunjaSqlite = { ... }: {
+      services.vikunja = {
+        enable = true;
+        database = {
+          type = "sqlite";
+        };
+        frontendScheme = "http";
+        frontendHostname = "localhost";
+      };
+      services.nginx.enable = true;
+    };
+    vikunjaPostgresql = { pkgs, ... }: {
+      services.vikunja = {
+        enable = true;
+        database = {
+          type = "postgres";
+          user = "vikunja";
+          database = "vikunjadb";
+        };
+        frontendScheme = "http";
+        frontendHostname = "localhost";
+        environmentFiles = [
+          (pkgs.writeText "vikunja-env" ''
+            VIKUNJA_DATABASE_PASSWORD=snakeoilpwd
```

Comment:
Might be nicer to go passwordless and connect through Unix sockets: https://github.com/jtojnar/nixfiles/blob/28a348c486e749f15be23f59e7cbea8efb08378f/hosts/azazel/ogion.cz/todo/default.nix#L59

Reply:
No, that wouldn't be nicer. The reason for that is that the module creates a systemd service with a dynamic user. As far as I know postgresql peer authentication is not possible with dynamic users.

Reply:
I never got around to testing if `DynamicUser` works with socket authentication on `postgresql` but it **does** with `mariadb`. You tested and confirms it does not?

Reply:
I tested (again) and it seems to work (requirement: database user has to be named the same as systemd service). So I changed the test to passwordless login.